src/openrct2/ride/Ride.cpp

```diff
@@ -3509,8 +3509,7 @@ static void ride_shop_connected(Ride* ride)
     }
 
     uint8_t tile_direction = trackElement->GetDirection();
-    entrance_directions <<= tile_direction;
-    entrance_directions = ((entrance_directions >> 12) | entrance_directions) & 0xF;
```

Comment:
The `>>12` is because the original code copied `bl` into `bh` but this implementation messed this up. A correct translation would have been ` ((entrance_directions >> 4) | entrance_directions) & 0xF;`